plugins/+matlab/+ciplugins/+github/BuildTaskGroupPlugin.m

```diff
@@ -0,0 +1,22 @@
+classdef BuildTaskGroupPlugin < matlab.buildtool.plugins.BuildRunnerPlugin
+
+%   Copyright 2024 The MathWorks, Inc.
+
+    methods (Access=protected)
+
+        function runTask(plugin, pluginData)
+           % Add Github worflow command for grouping the tasks
+           disp("::group::" + pluginData.TaskResults.Name + " - task");
+
+           runTask@matlab.buildtool.plugins.BuildRunnerPlugin(plugin, pluginData);
+
+           % Add Github workflow command ::error:: if the task is failed
+           if pluginData.TaskResults.Failed
+              disp("::error::" + pluginData.TaskResults.Name + " - task");
+           end
```

Comment:
It seems strange to have these errors that don't contain any diagnostic message. I realize this provides a link to the location in the log where above you might have additional context (not necessarily true on None verbosity, for example), but it seems out-of-model with the other errors I am used to in GitHub where the error provides a brief message about what went wrong.

To get access to the actual error message so we can put it in here, the plugin needs to listen for events similar to the `DiagnosticsOutputPlugin`. This is a little more involved than just overriding `runTask`, so we might want to consider it for a future iteration. Until then, maybe we don't put out these errors.

The name of this plugin may be too narrow as well. `BuildTaskGroupPlugin` indicates it solely focuses on putting out the group messages. But it is also putting out these error messages. If we intend this class to handle putting out all the messages for GitHub, a more general name might be more accurate. Some ideas: `GitHubActionsPlugin`, `GitHubWorkflowCommandsPlugin`, `WorkflowCommandsPlugin`.

Reply:
+1 - would be really nice if we could link directly to the error text being printed by the diagnostics plugin.

Reply:
Sure i will pick up the `WorkflowCommandsPlugin` just avoid GitHub as redundant word